internal/db/remote/commit/commit.go

```diff
@@ -99,6 +99,36 @@ func run(p utils.Program, ctx context.Context, username, password, database stri
 	}
 	defer conn.Close(context.Background())
 
+	// 1. Assert `supabase/migrations` and `schema_migrations` are in sync.
+	if err := AssertRemoteInSync(ctx, conn, fsys); err != nil {
+		return err
+	}
+
+	timestamp := utils.GetCurrentTimestamp()
+
+	// 2. Special case if this is the first migration
+	if localMigrations, err := afero.ReadDir(fsys, utils.MigrationsDir); err == nil && len(localMigrations) == 0 {
```

Comment:
It would have been caught and thrown in `AssertRemoteInSync` which reads the same directory. I'm assuming we don't run into race conditions with local fs between these calls, but can add more checks if it's safer.

Reply:
Yeah it's not obvious from a quick read (it's not in `AssertRemoteInSync`'s contract) but not a big deal